pliers/extractors/models.py

```diff
@@ -154,10 +156,21 @@ def _preprocess(self, stim):
             x = resizer.transform(stim).data
         else:
             x = stim.data
+
         if self.rescale_rgb:
+            if self.input_dtype.dtype.is_integer:
+                logging.warning('Rescaling the input from [0, 255] to [0, 1] and settings '
+                                'input_dtype to an integer type may result in loss of precision.'
+                                'Please, set rescale_rgb to False.')
+
             x = x / 255.0
-        x = tf.convert_to_tensor(x, dtype=tf.float32)
+
+        x = tf.convert_to_tensor(x, dtype=self.input_dtype)
         x = tf.expand_dims(x, axis=0)
+
+        if self.transform_inp:
```

Comment:
So you're just saying that this functionality should have been copied over from the parent `TFHubExtractor` class but wasn't?

Is there a way to do this with `super` to make this cleaner/more efficient?

Reply:
I believe it should- it provides some flexibility to adapt the input to the different requirements of TFHub models.

Using super would require the `x` to be a subclass of `Stim` given it uses the accessor `.data`, but it is a TF tensor at this point and it may be not too performatic to do the conversion.